lib/model/Capability.js

```diff
@@ -4,19 +4,35 @@ const Color = require('color');
 const Range = require(path.join(__dirname, 'Range.js'));
 
 module.exports = class Capability {
-  constructor(jsonObject) {
+  constructor(jsonObject, channel) {
     this.jsonObject = jsonObject; // calls the setter
+    this._channel = channel;
   }
 
   set jsonObject(jsonObject) {
     this._jsonObject = jsonObject;
+    this._cache = {};
   }
 
 
   get range() {
     return new Range(this._jsonObject.range); // required
   }
 
+  get ranges() {
+    if (!('ranges' in this._cache)) {
+      this._cache.ranges = [];
+      for (let i = this._channel.maxFineness; i >= 0; i--) {
```

Comment:
For `maxFineness=1`, you iterate 2 times through the loop. I guess this was not intended?